weave/trace/serialization/serializer.py

```diff
@@ -36,14 +36,54 @@ def load_instance(
 
 from __future__ import annotations
 
+import inspect
 from dataclasses import dataclass
-from typing import Any, Callable
+from typing import Any, Callable, Union
+
+from typing_extensions import TypeGuard
+
+# Not locking down the return type for inline encoding but
+# it would be expected to be something like a str or dict.
+InlineSave = Callable[[Any], Any]
+# Second parameter should really be MemTraceFilesArtifact
+# but we are avoiding a circular import.
+FileSave = Callable[[Any, Any, str], None]
```

Comment:
You can do an `if TYPE_CHECKING: import ...` for this

Reply:
Like this? (With an explanatory comment)
```
if TYPE_CHECKING:
    from weave.trace.serialization.mem_artifact import MemTraceFilesArtifact

    FileSave = Callable[[Any, MemTraceFilesArtifact, str], None]
else:
    FileSave = Callable[[Any, Any, str], None]
 ```

Reply:
I believe you can do this:

```py
if TYPE_CHECKING:
    from weave.trace.serialization.mem_artifact import MemTraceFilesArtifact

FileSave = Callable[[Any, "MemTraceFilesArtifact", str], None]
```